presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreen.kt

```diff
@@ -187,6 +187,34 @@ fun EventScreen(
         }
     )
 }
+private fun setupAlarmReminderItem(eventScreenState: EventScreenState, item: Int, context: Context): AlarmItem? {
+    /*
+       check what reminder that user has selected.
+       If user has set the following 10 minutes, 30 minutes, 1 hour, 6 hours. Set the alarm to go off before selected time
+       If user has set the following 1 day then use the date to set the alarm 1 day before
+    **/
+    return when(item) {
+        in 0..3 -> {
+            val startTime = eventScreenState.startTime
+
+            AlarmItem(
+                dateTime = startTime.minusMinutes(startTime.toEpochSecond()),
+                message = context.getString(eventScreenState.alarmReminderText)
+            )
+        }
+        4 -> {
+            val startDate = eventScreenState.startDate
+
+            AlarmItem(
+                dateTime = startDate.minusDays(startDate.toEpochSecond()),
+                message = context.getString(eventScreenState.alarmReminderText)
```

Comment:
@philipplackner @MKiperszmid The context is passed in here so I can use getString. This was to avoid the `Functions which invoke @Composable functions must be marked with the @Composable annotation`

Reply:
Yea that's fine

But regarding this code. All this logic shouldn't happen on the UI

You should calculate the time on the ViewModel at least

Reply:
Yes, the reason we do that in the ViewModel is that we then have a single testable source of truth for our state. If another UI component starts depending on it and you do it individually for all components on the UI every time, it can quickly cause bugs when requirements change